contribs/gmf/src/d3-lidar-profile/lidarprofileUtils.js

```diff
@@ -0,0 +1,487 @@
+
+  const lineStyle = new ol.style.Style({
+    stroke: new ol.style.Stroke({
+      color: 'rgba(255,0,0,1)',
+      width: widthInMapsUnits,
+      lineCap: 'square'
+    })
+  });
+
+  let firstSegmentAngle = 0;
+  let lastSegementAngle = 0;
+  const segNumber = clippedLine.getCoordinates().length - 1;
+  let segCounter = 1;
+
+  clippedLine.forEachSegment((start, end) => {
+    if (segCounter == 1) {
+      const dx = end[0] - start[0];
+      const dy = end[1] - start[1];
+      firstSegmentAngle = Math.atan2(dx, dy);
+    }
+
+    if (segCounter == segNumber) {
+      const dx = end[0] - start[0];
+      const dy = end[1] - start[1];
+
+      lastSegementAngle = Math.atan2(dx, dy);
+    }
+    segCounter += 1;
+
+
+  });
+
+  const styles = [lineStyle];
+  const lineEnd = clippedLine.getLastCoordinate();
+  const lineStart = clippedLine.getFirstCoordinate();
+
+  styles.push(
+    new ol.style.Style({
+      geometry: new ol.geom.Point(lineStart),
+      image: new ol.style.RegularShape({
+        fill: new ol.style.Fill({
+          color: 'rgba(255, 0, 0, 1)'
+        }),
+        stroke: new ol.style.Stroke({
+          color: 'rgba(255,0,0,1)',
+          width: 1,
+          lineCap: 'square'
+        }),
+        points: 3,
+        radius: 5,
+        rotation: firstSegmentAngle,
+        angle: Math.PI / 3
+      })
+    }),
+    new ol.style.Style({
+      geometry: new ol.geom.Point(lineEnd),
+      image: new ol.style.RegularShape({
+        fill: new ol.style.Fill({
+          color: 'rgba(255, 0, 0, 1)'
+        }),
+        stroke: new ol.style.Stroke({
+          color: 'rgba(255,0,0,1)',
+          width: 1,
+          lineCap: 'square'
+        }),
+        points: 3,
+        radius: 5,
+        rotation: lastSegementAngle,
+        angle: 4 * Math.PI / 3
+      })
+    })
+  );
+
+  return {
+    clippedLine: clippedLine.getCoordinates(),
+    distanceOffset: dLeft,
+    bufferGeom: feat,
+    bufferStyle: styles
+  };
+
+};
+
+/**
+* Get a LOD and with for a given chart span
+* Configuration is set up in Pytree configuration
+* @param {number} span domain extent
+* @return {{maxLOD: number, width: number}} Object with optimized LOD and width for this profile span
+*/
+gmf.lidarProfile.utils.prototype.getNiceLOD = function(span) {
+  let maxLOD = 0;
+  let width;
+  const levels = this.options.profileConfig.maxLevels;
+  for (const key in levels) {
+    if (span < key && levels[key].max > maxLOD) {
+      maxLOD = levels[key].max;
+      width = levels[key].width;
+    }
+  }
+  return {
+    maxLOD,
+    width
+  };
+};
+
+/**
+* Create a fake link and make the browser download the referenced URL
+* @param {string} filename csv file name
+* @param {string} dataUrl fake url from which to download the csv file
+* @export
+*/
+gmf.lidarProfile.utils.prototype.downloadDataUrlFromJavascript = function(filename, dataUrl) {
+
+  const link = document.createElement('a');
```

Comment:
you should be able to use The saveAs function from FileSaver library, see: https://github.com/camptocamp/ngeo/blob/2.2/src/services/download.js and https://github.com/camptocamp/ngeo/blob/2.2/src/services/featurehelper.js#L844-L846